test/acceptance/test/core_operations_test.go

```diff
@@ -60,180 +61,310 @@ spec:
+func getRepoName() (string, error) {
+	name, err := getWegoRepoName()
+	return name + "-" + filepath.Base(tmpDir), err
 }
 
-func setUpTestRepo(t *testing.T) {
+func setUpTestRepo() error {
 	dir, err := os.Getwd()
-	require.NoError(t, err)
-	require.NoError(t, os.Chdir(tmpDir))
-	defer func() {
-		require.NoError(t, os.Chdir(dir))
-	}()
+	if err != nil {
+		return err
+	}
+	err = os.Chdir(tmpDir)
+	if err != nil {
+		return err
+	}
+
 	_, err = utils.CallCommand("git init")
-	require.NoError(t, err)
-	_, _ = utils.CallCommand("git checkout -b main")
+	if err != nil {
+		return err
+	}
+
 	err = ioutil.WriteFile("nginx.yaml", []byte(nginxDeployment), 0666)
-	require.NoError(t, err)
+	if err != nil {
+		return err
+	}
+
 	err = utils.CallCommandForEffectWithDebug("git add nginx.yaml && git commit -m'Added workload'")
-	require.NoError(t, err)
-	cloneurl := fmt.Sprintf("https://github.com/%s/%s", os.Getenv("GITHUB_ORG"), getRepoName(t))
+	if err != nil {
+		return err
+	}
+
+	name, err := getRepoName()
+	if err != nil {
+		return err
+	}
+
+	cloneurl := fmt.Sprintf("https://github.com/%s/%s", os.Getenv("GITHUB_ORG"), name)
 	ref, err := gitprovider.ParseOrgRepositoryURL(cloneurl)
-	require.NoError(t, err)
+	if err != nil {
+		return err
+	}
+
 	ctx := context.Background()
 	_, err = client.OrgRepositories().Create(ctx, *ref, gitprovider.RepositoryInfo{
 		Description: gitprovider.StringVar("test repo"),
 	}, &gitprovider.RepositoryCreateOptions{
 		AutoInit:        gitprovider.BoolVar(true),
 		LicenseTemplate: gitprovider.LicenseTemplateVar(gitprovider.LicenseTemplateApache2),
 	})
+	if err != nil {
+		return err
+	}
+
+	name, err = getRepoName()
+	if err != nil {
+		return err
+	}
+
+	originurl := fmt.Sprintf("https://github.com/%s/%s", os.Getenv("GITHUB_ORG"), name)
+	err = utils.CallCommandForEffectWithDebug(fmt.Sprintf("git remote add origin %s && git pull --rebase origin main && git checkout main && git push --set-upstream origin main", originurl))
+	if err != nil {
+		return err
+	}
+
+	return os.Chdir(dir)
+}
 
-	require.NoError(t, err)
-	originurl := fmt.Sprintf("ssh://git@github.com/%s/%s", os.Getenv("GITHUB_ORG"), getRepoName(t))
-	cmd := fmt.Sprintf(`git remote add origin %s && \
-		git pull --rebase origin main && \
-		git push --set-upstream origin main`,
-		originurl)
-	err = utils.CallCommandForEffectWithDebug(cmd)
-	require.NoError(t, err)
+func runCommandForGinkgo(cmd string) error {
+	command := exec.Command("sh", "-c", utils.Escape(cmd))
+	session, err = gexec.Start(command, GinkgoWriter, GinkgoWriter)
+	return err
 }
 
-func deleteRepos(t *testing.T) {
+func deleteRepos() error {
 	clusterName, err := status.GetClusterName()
 	if err == nil {
 		ctx := context.Background()
-		url := fmt.Sprintf("https://github.com/%s/%s", os.Getenv("GITHUB_ORG"), getRepoName(t))
+		name, err := getRepoName()
+		if err != nil {
+			return err
+		}
+
+		url := fmt.Sprintf("https://github.com/%s/%s", os.Getenv("GITHUB_ORG"), name)
```

Comment:
SUGGESTION: As it may be GitLab or some privately hosted git in the future, I would suggest not to hard code the URLs.